File: backend/GeoChatroom.py
from Message import Message
from Chatroom import Chatroom

class GeoChatroom(Chatroom):
    def __init__(self, id, latitude, longitude, radius):
        super().__init__(id) 
        self.latitude = latitude
        self.longitude = longitude
        self.radius = radius

    def getType(self):
        return "geo"
        
    def toDict(self):
        return {
            "chatId": self.id,
            "latitude": self.latitude,
            "longitude": self.longitude,
            "radius": self.radius,
            "messages": len(self.messages),
            "type": self.getType()
        }

File: android/app/src/main/java/pt/ulisboa/tecnico/cmov/conversationalist/data/backend/responses/GeoChatroom.java
package pt.ulisboa.tecnico.cmov.conversationalist.data.backend.responses;

public class GeoChatroom extends Chatroom {
    private final double latitude;
    private final double longitude;

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public double getRadius() {
        return radius;
    }

    private final double radius;

    public GeoChatroom(String chatId, String type, int messages, double latitude, double longitude, double radius) {
        super(chatId, type, messages);
        this.latitude = latitude;
        this.longitude = longitude;
        this.radius = radius;
    }
}
